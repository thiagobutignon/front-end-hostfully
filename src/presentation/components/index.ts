export { default as ErrorComponent } from './error/error'
export { default as HeaderComponent } from './header/header'
export { default as InputError } from './input/input-error'
export { default as InputWithLabel } from './input/input-with-label'
export { default as LayoutComponent } from './layout/layout'
export { default as PropertyInfoComponent } from './property-info/property-info'
export { default as PropertyCardComponent } from './select/property-select'
export { default as ThemeModeComponent } from './theme-mode/theme-mode'
