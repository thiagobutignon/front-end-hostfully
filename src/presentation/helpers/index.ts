export * from './populate-form-fields'
export * from './wrap-chakra'
