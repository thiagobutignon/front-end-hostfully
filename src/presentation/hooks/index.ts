export * from './booking'
export * from './error'
export * from './property'
