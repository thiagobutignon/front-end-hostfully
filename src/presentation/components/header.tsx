import { Box, Flex, Menu, Stack, Text } from '@chakra-ui/react'

import { ThemeModeComponent } from '@/presentation/components'

const HeaderComponent: React.FC = () => {
  return (
    <>
      <Box bg="reverseBackground" px={4}>
        <Flex h={16} alignItems={'center'} justifyContent={'space-between'}>
          <Text color="reverseText" fontSize="xl" fontWeight={700}>
            Hostfully
          </Text>

          <Flex alignItems={'center'}>
            <Stack direction={'row'}>
              <Menu>
                <ThemeModeComponent />
              </Menu>
            </Stack>
          </Flex>
        </Flex>
      </Box>
    </>
  )
}

export default HeaderComponent
