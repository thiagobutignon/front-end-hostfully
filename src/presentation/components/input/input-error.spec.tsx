import '@testing-library/jest-dom'

import { RenderResult, render } from '@testing-library/react'

import { FormControl } from '@chakra-ui/react'
import { InputError } from '@/presentation/components'

const makeSut = (props?: any): RenderResult => {
  return render(
    <FormControl>
      <InputError {...props}>Error message</InputError>
    </FormControl>
  )
}

describe('InputError', () => {
  test('renders InputError correctly', () => {
    const { getByText } = makeSut()

    expect(getByText('Error message')).toBeInTheDocument()
  })

  test('applies the correct styles', () => {
    const { getByText } = makeSut()

    const inputErrorElement = getByText('Error message')

    expect(inputErrorElement).toHaveStyle({ color: 'error' })
  })

  test('matches snapshot', () => {
    const { asFragment } = render(
      <FormControl>
        <InputError>Error message</InputError>
      </FormControl>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
