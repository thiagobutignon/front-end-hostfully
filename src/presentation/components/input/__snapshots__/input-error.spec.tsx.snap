// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputError matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-form-control css-0"
    role="group"
  >
    <div
      class="chakra-form__helper-text css-18w01t4"
      id="field-:r2:-helptext"
    >
      Error message
    </div>
  </div>
</DocumentFragment>
`;
